### PR TITLE
dynamic_hook: cumulative changes

### PR DESCRIPTION
**Pull request by ADD-SP**
# REBASE AND MERGE

### Summary

Please review commit-by-commit. This PR did some style cleaning, fixed some bugs, and removed unused features.

### Checklist

- [X] The Pull Request has tests
- [N/A] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [N/A] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

_[KAG-4590]_

[KAG-4590]: https://konghq.atlassian.net/browse/KAG-4590?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: kong/dynamic_hook/init.lua [hunk 1]
```diff
@@ -213,6 +212,9 @@ end
 
 
 function _M.run_hooks(group_name, hook_name, a1, a2, a3, a4, a5, a6, a7, a8, ...)
```
Comment (samugi) [inline kong/dynamic_hook/init.lua:214]: should we rename this to: `run_hook`? It looks like it's running only one

Comment (ADD-SP) [reply kong/dynamic_hook/init.lua]: done

**Review by samugi — APPROVED** [review]

File: kong/dynamic_hook/init.lua [hunk 2]
```diff
@@ -132,37 +129,37 @@ local function execute_original_func(max_args, original_func, a1, a2, a3, a4, a5
...
+      execute_hooks(handlers.afters, "after", group_name, r1, r2, r3, r4, r5, r6, r7, r8)
     end
-    return a1, a2, a3, a4, a5, a6, a7, a8
+
+    return r1, r2, r3, r4, r5, r6, r7, r8
   end
 end
 
 
 function _M.hook_function(group_name, parent, child_key, max_args, handlers)
```
[the remark this answers is not in the document]
Comment (ADD-SP) [reply kong/dynamic_hook/init.lua]: Good point! Fixed.

File: kong/dynamic_hook/init.lua [hunk 3]
```diff
@@ -132,48 +129,48 @@ local function execute_original_func(max_args, original_func, a1, a2, a3, a4, a5
...
   end
 end
 
 
-function _M.hook_function(group_name, parent, child_key, max_args, handlers)
+function _M.hook_function(group_name, parent, function_key, max_args, handlers)
+  assert(type(group_name) == "string", "group_name must be a string")
   assert(type(parent) == "table", "parent must be a table")
-  assert(type(child_key) == "string", "child_key must be a string")
+  assert(type(function_key) == "string", "child_key must be a string")
```
Comment (samugi) [inline kong/dynamic_hook/init.lua:159]: forgot one:
```suggestion
  assert(type(function_key) == "string", "function_key must be a string")
```

Comment (ADD-SP) [reply kong/dynamic_hook/init.lua]: Fixed

**Review by chronolaw — APPROVED** [review]

**Pull request merged**